app/views/items/_item.html.erb

```diff
@@ -1,5 +1,6 @@
 <% @items.each do |item| %>
   <ul>
-    <li><%= item.name %></li>
+    <li><%= link_to "", @item, method: :delete, class: 'gylphicon glyphicon-ok' %>
```

Comment:
this is the link you're having trouble with, right?
the `link_to` method takes 2 arguments (and then a hash of other optional arguments):
1. the text shown for the link (in this case there is no text and just the icon displayed by the class)
2. the url where the link should go to

It's argument 2 that is causing the problem.  You're using a rails shorthand which is to pass in an object that rails will then interpret into a link (e.g. if you pass in an item object with id of 5, rails will figure out that the url should be "/items/5").  You're passing `@item`.  What's the value of `@item`?